clarity_ext/service/dilution/service.py

```diff
@@ -431,6 +440,54 @@ def __repr__(self):
             tuple(self.update_info))
 
 
+class VirtualTransferBatch(object):
+    """Contains the original "virtual" transfers we want performed. These will translate into
+    a series of actual SingleTransfer objects."""
+
+    def __init__(self, transfers):
+        """Creates a virtual batch and updates each transfer so it has a pointer to the it"""
+        self.virtual_transfers = VirtualTransferBatch.transfers_by_output(transfers)
+        self._transfers = transfers
+
+        for transfer in transfers:
+            transfer.virtual_batch = self
+
+    def __getitem__(self, transfer):
+        """Returns the virtual transfer by the transfer"""
+        return self.virtual_transfers[transfer.target_location.artifact.id]
+
+    @staticmethod
+    def transfers_by_output(transfers):
```

Comment:
Maybe add a comment of what is the dict-key? It takes time to penetrate the return statement.

Reply:
:+1: